pkg/cmd/cli/cmd/rsync.go

```diff
@@ -0,0 +1,381 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"gopkg.in/pipe.v2"
+	kclient "k8s.io/kubernetes/pkg/client"
+	kubecmd "k8s.io/kubernetes/pkg/kubectl/cmd"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	// RsyncRecommendedName is the recommended name for the rsync command
+	RsyncRecommendedName = "rsync"
+
+	rsyncLong = `
+Copy local files to a container
+
+This command will attempt to copy local files to a remote container. It will default to the
+first container if none is specified, and will attempt to use 'rsync' if available locally and in the
+container. If not present, it will attempt to use 'tar' to send files to the container.`
```

Comment:
After talking to @bparees at least locally you wouldn't need an external program. We can tar the directory for you in go and stream it into the pod. You'd only need the tar in the container.